src/components/PeopleFilters.tsx

```diff
@@ -1,23 +1,58 @@
+import { useSearchParams } from 'react-router-dom';
+import cn from 'classnames';
+import { Sex } from '../types/Sex';
+import { centuryButtons } from '../utils/peopleHelper';
+import { getSearchWith, SearchParams } from '../utils/searchHelper';
+import { SearchLink } from './SearchLink';
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex') || 'All';
+  const centuries = searchParams.getAll('centuries') || [];
+
+  const setSearchWith = (params: SearchParams) => {
+    const search = getSearchWith(searchParams, params);
+
+    setSearchParams(search);
+  };
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchWith({ query: event.target.value });
+  };
+
   return (
     <nav className="panel">
       <p className="panel-heading">Filters</p>
 
       <p className="panel-tabs" data-cy="SexFilter">
-        <a className="is-active" href="#/people">
+        <SearchLink
+          className={cn({ 'is-active': sex === 'All' })}
```

Comment:
Let's move `All` to const or enum and reuse, I see at least few places where we can use it